ipa-core/src/protocol/dp/mod.rs

```diff
@@ -21,6 +21,36 @@ use crate::{
         BitDecomposed, FieldSimd, TransposeFrom, Vectorizable,
     },
 };
+
+/// Struct to hold noise parameters, contains internal values not received from the client
+pub struct NoiseParams {
+    pub epsilon: f64,
+    pub delta: f64,
+    pub per_user_credit_cap: f64,
+    pub success_prob: f64,
```

Comment:
I would prefer to see compile safety here or at least enforce at construction time that this value is within 0..1 range. You do plenty of math down below and it can underflow easily if this value is outside of that range. We don't want to deal with that in remote servers

Reply:
added some more validation checks.